x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/ToCharTestScript.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    private static <T> T randomFromCollection(Collection<T> list) {
+        List<T> l = new ArrayList<>(list);
+        return l.get(rnd().nextInt(l.size()));
+    }
+
+    private static String patternWithRandomModifiers(String pattern) {
+        if (NOT_FULLY_MATCHABLE_PATTERNS.contains(pattern)) {
+            return pattern;
+        }
+        return randomFromCollection(FILL_MODIFIERS) + pattern + randomFromCollection(ORDINAL_SUFFIX_MODIFIERS);
+    }
+
+    private static String adjustZoneIdToPostgres(String zoneId) {
+        // when the zone is specified by the offset in Postgres, it follows the POSIX definition, so the +- signs are flipped
+        // compared to ISO-8601, see more info at: https://www.postgresql.org/docs/current/datetime-posix-timezone-specs.html
+        if (zoneId.startsWith("+")) {
+            zoneId = zoneId.replaceFirst(quote("+"), "-");
+        } else if (zoneId.startsWith("-")) {
+            zoneId = zoneId.replaceFirst(quote("-"), "+");
+        }
+        return zoneId;
+    }
+
+    /**
+     * Generates an SQL file (psql input) that can be used to create the test dataset for the unit test of the <code>TO_CHAR</code>
+     * implementation. In case the <code>TO_CHAR</code> implementation needs an upgrade, add the list of the new format
+     * strings to the list of the format string, regenerate the SQL, run it against the PostgreSQL version you are targeting
+     * and update the test CSV file.
+     */
+    private String unitTestExporterScript() {
+        String header =
+            "\n\\echo #" +
+            "\n\\echo # DO NOT EDIT manually, was generated using " + ToCharTestScript.class.getName() +
+            "\n\\echo #\n\n";
+        String testCases = testRecords.stream().map(tc -> {
+            long seconds = tc.secondsAndFractionsSinceEpoch.longValue();
+            BigDecimal fractions = tc.secondsAndFractionsSinceEpoch.remainder(BigDecimal.ONE).movePointRight(6);
+            return String.format(Locale.ROOT, 
+                "SET TIME ZONE '%6$s';\n"
+                    + "\\copy (SELECT %1$s as epoch_seconds_and_microsends, '%5$s' as zone_id, '%4$s' as format_string, " 
+                    + "(TO_TIMESTAMP(%2$d) + INTERVAL '%3$d microseconds') as to_timestamp_result, "
+                    + "TO_CHAR((TO_TIMESTAMP(%2$d) + INTERVAL '%3$d microseconds'), '%4$s') as to_char_result) to stdout " 
+                    + "with DELIMITER as '" + DELIMITER + "' NULL as '' csv \n",
+                tc.secondsAndFractionsSinceEpoch.toPlainString(),
+                seconds,
+                fractions.intValue(),
+                tc.formatString,
+                tc.zoneId,
+                adjustZoneIdToPostgres(tc.zoneId));
+        }).collect(Collectors.joining("\n"));
+        return header + testCases;
+    }
+
+    public static void main(String[] args) throws Exception {
+        String scriptFilename = args.length < 1 ? "/tmp/postgresql-tochar-test.sql" : args[0];
```

Comment:
Store the result in current directory, it's much more common - `/tmp` is not portable nor something where folks will typically look.